torch/distributed/device_mesh.py

```diff
@@ -65,17 +65,18 @@ def _init_device_mesh_stub():
                 "DeviceMesh requires numpy >= 1.21 to be installed for type checking"
             )
 
+    BackendConfig = tuple[Optional[str], Optional[C10dBackend.Options]]
+    torch.serialization.add_safe_globals([_MeshLayout])
+
     class _MeshEnv(threading.local):
         def __init__(self) -> None:
             self.mesh_stack: list[DeviceMesh] = []
             self.child_to_root_mapping: dict[DeviceMesh, DeviceMesh] = {}
-            self.mesh_dim_group_options: dict[
-                int, tuple[Optional[str], Optional[C10dBackend.Options]]
-            ] = {}
+            self.mesh_dim_group_options: dict[int, BackendConfig] = {}
```

Comment:
Just for reference, this is something you could put in an earlier stacked PR, easy BE and easy stamp

Reply:
Split and merged in https://github.com/pytorch/pytorch/pull/163928